hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ClobTypeDescriptor.java

```diff
@@ -129,6 +139,9 @@ else if ( Reader.class.isAssignableFrom( value.getClass() ) ) {
 			Reader reader = (Reader) value;
 			return options.getLobCreator().createClob( DataHelper.extractString( reader ) );
 		}
+		else if ( String.class.isAssignableFrom( value.getClass() ) ) {
```

Comment:
Should we use `String.class.isAssignableFrom ( value )` instead?

Reply:
No, the `isAssignableFrom` method requires an argument of type `Class<?>`. See lines 134 and 138 also.